Framework/Kernel/src/VectorHelper.cpp

```diff
@@ -533,6 +533,166 @@ void linearlyInterpolateY(const std::vector<double> &x, std::vector<double> &y,
     step++;
   }
 }
+namespace {
+/** internal function converted from Lambda to identify interval around
+*specified
+* point and  run average around this point
+*@param index      -- index to average around
+*@param startIndex -- index in the array of data (input to start average
+*                     from) should be: index>=startIndex>=0
+*@param endIndex   -- index in the array of data (input to end average at)
+*                     should be: index<=endIndex<=input.size()
+*@param halfWidth  -- half width of the interval to integrate.
+*@param input      -- vector of input signal
+*@param binBndrs   -- pointer to vector of bin boundaries or NULL pointer.
+ */
+double runAverage(size_t index, size_t startIndex, size_t endIndex,
+                  const double halfWidth, const std::vector<double> &input,
+                  std::vector<double> const *const binBndrs) {
```

Comment:
Does binDirs have to be a pointer? Can it be a const ref? The function's code would look clearer without ->operator[](...)